### PR TITLE
Paint pot

### PR DESCRIPTION
**Pull request by amyzhaang**

### CODE REVIEW

File: appinventor/codelabs/paint-pot/index.html [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+<body unresolved class="fullbleed">
+
+  <google-codelab title="Paint Pot Codelab" feedback-link="https://goo.gl/forms/9zDi1V3arg167lIY2">
+
+      <google-codelab-step label="Overview">
+        <p>In this codelab, you&#39;ll make an app to draw on your phone.</p>
+        <h3 class="checklist">What you&#39;ll learn</h3>
+        <ul class="checklist">
+          <li>Use the <b>Canvas</b> component for drawing.</li>
+          <li>Learn how to use <b>local variables</b> and <b>global variables</b>.</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:30]: Learn how to use <b>variables</b> to store information.

File: appinventor/codelabs/paint-pot/index.html [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+          <li>Work with different <b>event handlers</b> and types of blocks.</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:31]: Use **touch** and **drag** event handlers with the Canvas component.

File: appinventor/codelabs/paint-pot/index.html [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      <google-codelab-step label="Create a new project">
+      <p>Create a new project and name it "PaintPot"</p>
+      <img src="img/new_project.png">
+      <p>In the <b>Properties</b> panel, change the <b>title</b> to "PaintPot"</p>
+      <img src="img/screen_title.png" style="max-width: 400px">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add color buttons">
+      <p>Drag three <b>Button</b> components into your workspace. Name the button components "ButtonRed","ButtonBlue", and "ButtonGreen".</p>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:45]: Add space after "ButtonRed"

(Also, you could consider using `<code>` tags)

File: appinventor/codelabs/paint-pot/index.html [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        <ol>
+          <li>Make your buttons look as shown below.</li>
+          <img src="img/color_buttons.png">
+        </ol>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add a canvas and wipe button">
+      <p>From the <b>Drawing and Animation</b> drawer, drag a <b>Canvas</b> component onto the workspace. Change the <b>name</b> of the component to "DrawingCanvas", set the <b>Width</b> to "Fill Parent", and set the <b>Height</b> to 300 pixels.</p>
+      <p>Download kitty.png from <a href="http://examples.oreilly.com/0636920016632/" target="_blank">here</a>. Upload the file in the <b>Media</b> panel.</p>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:57]: Use https://resources.oreilly.com/examples/0636920016632/raw/master/kitty.png instead

File: appinventor/codelabs/paint-pot/index.html [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+      <p>Let's change the background image of the canvas to this picture so we can draw on it. In the <b>Properties</b> panel of the Canvas component, set the <b>BackgroundImage</b> to kitty.png.</p>
+      <img src="img/canvas_background.png">
+      <p>Drag a <b>Button</b> component below the Canvas. Name the component "ButtonWipe".</p>
+      <p>Your app should look like this:</p>
+      <img src="img/part1_ui.png">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add button event handlers">
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:66]: button click event handlers

File: appinventor/codelabs/paint-pot/index.html [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+      <p>Now it's time to code our app. Go to <b>Blocks View</b>.</p>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:67]: ~Now it's time to code our app.~

File: appinventor/codelabs/paint-pot/index.html [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+        </ol>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Checkoff #1">
+      <aside class="special">
+          <h2>paint-pot-1</h2>
+          <p><b>To receive this checkoff, you must</b></p>
+          <ul class="checklist">
+            <li>Demonstrate your working app</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:102]: - Show that you can draw a point by touching the canvas and a line by dragging your finger across the canvas.
- Show that you can change the color of the paint by clicking on the different colored buttons.

File: appinventor/codelabs/paint-pot/index.html [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+            <li>Explain what local variables are (the orange arguments built into some event handlers)</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:103]: I would remove these two bullets that talk about the local variables. We don't mention it anywhere.

File: appinventor/codelabs/paint-pot/index.html [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+            <li>Explain the six local variables in the <code>when DrawingCanvas.Dragged</code> block (startX, startY, prevX, etc.)</li>
+            <li>Explain the difference between touching and dragging the screen</li>
+          </ul>
+        </aside>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Creating variables">
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:111]: Setting paint brush sizes

File: appinventor/codelabs/paint-pot/index.html [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+      <p>Now we're going to create a way to change the size of the dot you draw when you touch the screen. To do this, we're going to use <i>variables</i>.</p>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:112]: Include a `<aside class="note">` with a blurb about variables.

File: appinventor/codelabs/paint-pot/index.html [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+          <ul>
+            <li>a variable named "big" whose value is 8</li>
+            <li>a variable named "dotsize" whose value is 2</li>
+          </ul>
+        </ol>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add two more buttons">
+      <p>Go back to <b>Designer View</b>. Add a <b>Horizontal Arrangement</b> component to below the canvas component. Drag the Wipe button into the HorizontalArrangement.</p>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:132]: Add a <b>Horizontal Arrangement</b> component ~to~ below the canvas component.

File: appinventor/codelabs/paint-pot/index.html [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+      <p>Add two more button components into the HorizontalArrangement, next to the Wipe button. Name the buttons "ButtonBig" and "ButtonSmall", and set their <b>Text</b> to read "Big dots" and "Small dots", respectively.</p>
+      <img src="img/final_ui.png">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Changing the values of variables">
+      <p>Go to <b>Blocks View</b>. Find the <code>when DrawingCanvas.Touched</code> event handler and replace the number 5 with <code>get dotsize</code> from the <b>Variables</b> drawer.</p>
+      <img src="img/touch_variable.png" style="max-width: 400px">
+      <p>Drag a <code>when ButtonSmall.Clicked</code> event handler into the workspace. Place a <code>get global dotsize to</code> block in the event handler, and plug in <code>get global small</code> into the block.</p>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:141]: <code>set global dotsize to</code>

File: appinventor/codelabs/paint-pot/index.html [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+      <img src="img/final_blocks.png">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Checkoff #2">
+      <aside class="special">
+          <h2>paint-pot-2</h2>
+          <p><b>To receive this checkoff, you must</b></p>
+          <ul class="checklist">
+            <li>Demonstrate your working app</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:154]: Show that you can change the size of the dots.

File: appinventor/codelabs/paint-pot/index.html [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+            <li>Explain what variables are</li>
+            <li>Explain how variables were used to change dot size</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:156]: Explain what variables are and you used them to change the dot size

File: appinventor/codelabs/paint-pot/index.html [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Checkoff #2">
+      <aside class="special">
+          <h2>paint-pot-2</h2>
+          <p><b>To receive this checkoff, you must</b></p>
+          <ul class="checklist">
+            <li>Demonstrate your working app</li>
+            <li>Explain what variables are</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:155]: Remove this one (combined with bullet below)

File: appinventor/codelabs/paint-pot/index.html [hunk 16]
```diff
@@ -0,0 +1,221 @@
...
+            <li>Explain how variables were used to change dot size</li>
+            <li>Explain the difference between local variables and global variables</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:157]: Remove this one

File: appinventor/codelabs/paint-pot/index.html [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+          </ul>
+        </aside>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Congratulations!">
+        <p>You have used App Inventor to create a drawing app.</p>
+        <h3 class="checklist"><strong>What we&#39;ve covered</strong></h3>
+        <ul class="checklist">
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:166]: Update to be the same as the overview checklist

File: appinventor/codelabs/paint-pot/index.html [hunk 18]
```diff
@@ -0,0 +1,221 @@
...
+          <li>Learn how to use <b>local variables</b> and <b>global variables</b>.</li>
+          <li>Work with different <b>event handlers</b> and types of blocks.</li>
+        </ul>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Bonus Features">
+        If you finish early, implement one or more of the following features for a bonus checkoff.
+        <aside class="special">
+          <h2>talk-to-me-bonus</h2>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:177]: paint-pot bonus

File: appinventor/codelabs/paint-pot/index.html [hunk 19]
```diff
@@ -0,0 +1,221 @@
...
+          <p><b>To receive this checkoff, you must implement one or more of the following:</b></p>
+          <ul>
+            <li>Implement a way for the user to imput the size of the dot.</li>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:180]: input

**Review by kelseyc18 — CHANGES_REQUESTED** [review]

File: appinventor/codelabs/paint-pot/index.html [hunk 20]
```diff
@@ -0,0 +1,221 @@
...
+    }
+  </style>
+
+</head>
+<body unresolved class="fullbleed">
+
+  <google-codelab title="Paint Pot Codelab" feedback-link="https://goo.gl/forms/9zDi1V3arg167lIY2">
+
+      <google-codelab-step label="Overview">
+        <p>In this codelab, you&#39;ll make an app to draw on your phone.</p>
```
Comment (kelseyc18) [inline appinventor/codelabs/paint-pot/index.html:26]: Please add a screenshot of the final product.

File: appinventor/codelabs/paint-pot/index.html (hunk 3 above)
Comment (amyzhaang) [reply appinventor/codelabs/paint-pot/index.html:45]: hmm i've been saving the <code>&lt;code&gt;</code> tags for the actual code blocks, but if you want anything typed to be <code>&lt;code&gt;</code> then i can change it!

Comment (kelseyc18) [reply appinventor/codelabs/paint-pot/index.html:45]: Keeping it in quotes is fine :)

**Pull request merged**